app/Http/Middleware/TrustProxies.php

```diff
@@ -8,23 +8,16 @@
 class TrustProxies extends Middleware
 {
     /**
-     * The trusted proxies for this application,
-     * sourced from 'config/trustedproxy.php'.
+     * The trusted proxies for this application.
      *
      * @var array
      */
     protected $proxies;
 
     /**
-     * The current proxy header mappings.
+     * The headers that should be used to detect proxies.
      *
-     * @var array
+     * @var int
      */
-    protected $headers = [
-        Request::HEADER_FORWARDED => null, // Not set on AWS or Heroku.
-        Request::HEADER_X_FORWARDED_FOR => 'X_FORWARDED_FOR',
-        Request::HEADER_X_FORWARDED_HOST => null, // Not set on AWS or Heroku.
-        Request::HEADER_X_FORWARDED_PORT => 'X_FORWARDED_PORT',
-        Request::HEADER_X_FORWARDED_PROTO => 'X_FORWARDED_PROTO',
-    ];
+    protected $headers = Request::HEADER_X_FORWARDED_ALL;
```

Comment:
I'm less familiar with Trusted Proxy and want to make sure the new bit property is fine that should encompass the prior series of headers.

Reply:
I think we want to use the `Request::HEADER_X_FORWARDED_AWS_ELB` constant here (which, I think, like our previous manual settings here) would ignore the `Forwarded` and `X-Forwarded-Host` values that aren't used on Heroku or AWS...)

We might be fine with `_ALL`... but I'd just want to double-check that those extra headers can't be misused.

Reply:
Ah yep, here's Heroku's [documentation on how to configure this](https://devcenter.heroku.com/articles/getting-started-with-laravel#trusting-the-load-balancer) which suggests `Request:: HEADER_X_FORWARDED_AWS_ELB`:

> It is very important to also prevent Laravel from trusting the `Forwarded` and `X-Forwarded-Host` headers, because Heroku’s router does not set those, but Symfony’s Request component trusts them out of the box once a trusted proxy is set.

Reply:
Ah ok! Thanks for the info on this @DFurnes 🙇